pkg/deploy/controller/configchange/controller.go

```diff
@@ -112,29 +116,40 @@ func canTrigger(config *deployapi.DeploymentConfig) (bool, []deployapi.Deploymen
 	return ictCount == resolved, causes
 }
 
-// hasTemplateDiff compares the pod template of the latest deployment for the provided deployment config
-// against the current pod template of the deployment config. If there is a diff, the controller is supposed
-// to let the world know.
-func (c *DeploymentConfigChangeController) hasTemplateDiff(config *deployapi.DeploymentConfig) (bool, error) {
+// decodeFromLatest will try to return the previous version of the current deploymentconfig.
+// If there is no previous deploymentconfig (latestVersion == 0), the returned deploymetnconfig
+// will be the same.
+func (c *DeploymentConfigChangeController) decodeFromLatest(config *deployapi.DeploymentConfig) (*deployapi.DeploymentConfig, error) {
+	if config.Status.LatestVersion == 0 {
+		return config, nil
+	}
+
 	latestDeploymentName := deployutil.LatestDeploymentNameForConfig(config)
 	deployment, err := c.kClient.ReplicationControllers(config.Namespace).Get(latestDeploymentName)
 	if err != nil {
 		// If there's no deployment for the latest config, we have no basis of
 		// comparison. It's the responsibility of the deployment config controller
 		// to make the deployment for the config, so return early.
-		if kerrors.IsNotFound(err) {
-			glog.V(5).Infof("Ignoring change for deployment config %q; no existing deployment found", deployutil.LabelForDeploymentConfig(config))
-			return false, nil
-		}
-		return false, fmt.Errorf("couldn't retrieve deployment for deployment config %q: %v", deployutil.LabelForDeploymentConfig(config), err)
+		return nil, fmt.Errorf("couldn't retrieve deployment for deployment config %q: %v", deployutil.LabelForDeploymentConfig(config), err)
 	}
 
-	dc, err := c.decodeConfig(deployment)
-	if err != nil {
-		return false, fatalError(fmt.Sprintf("error decoding deployment config from deployment %q for deployment config %s: %v", deployutil.LabelForDeployment(deployment), deployutil.LabelForDeploymentConfig(config), err))
-	}
+	return c.decodeConfig(deployment)
```

Comment:
Note for future ref: I think this decoder abstraction has proven to be annoying/unnecessary and we can probably just replace it with a simple stateless utility.